dev/diagrams/diagrams.drawio

```diff
@@ -1 +1,647 @@
WSRJarpjiRFeIjGbj2MgbJZ0MdogjpOOU+H95Q3Pf/n83fJav3Z58rmVa2rGtY3mmWw6/eKK5VgmQL+sy2L6SrbwwYGWQXJIYucFpjarlx8h8e8Pk9CQvRjb5lVpTy44nlto+YmQ5OM0qauS6+X8xGlk6JdDEcKCSq0WmpBhX1BqFHAgxMqjmapHZEFOYopBo9vxft0/2y8d7PJ5efTBHv74+5WxJn29DcmDXI4JwHqLtSLJ9aythE0IhHoeh6zixy90U4MKQsrJIWaLJ2IcESpXtP+ZntMV04yMfZjHYTFArp4KUHMyc0HLHGcPmS2uWkWWxpLQmMFL58EXgtL+5R46f9hc/kqTyG++2xk/gVDR+siJf5fCbb7DcGT+BU9H4iRtUKoifpezL/gROReMnK+VVDj9h4/LW+AmcisZPVgysHH7q3vATOBWNn2zjzCb4wamLf5HfdeVM0XVG+E3D/DONXV1OWdQfXcxSF3cwcMkIaO4Q0TZThnShJC8LKovScKnHmsvJK/lYfHJStMrItsWfvMmrislFOvaWJs8zOjh+sn0FJ48f77K3tT7BY/OMCkYvUZYUel00HCI//49k/L13z2PgYxcTiZF+KGdn7wUdOGi9p4d8nFTt1z3EZPN/n/OmJ9le38xRwb1Xf7RKVn9UlfeA29qiwIjvTNHGqAn4RQXv67XWH95i+bSR3aUlO79BpjmF1U81sX4DyXBolzsghHFxWxmA5MSN+Z6pTU7deINA6dyanCo7aEO2eaA4pMRKzalaiarwhx4c8H9ccuFvuUR6CsI/5F+35MKv6rKoAMUhV6jlUIi5T1WhOPYStS4mMl/pbqPT9Ep2Mxs4yeZj2YkWxYlfTD1OWPyqwtdZjy7/qqUOggc6eu6gi7nDimN/FHasT0jySpw8GScRysPt5W0FJpK6yq8YNsQ5XWsUhGN4A0L96sPk9eLPBfoC1ODzY11yhF+ERYRKSN5BvhA9hBEP6M+19oqz7DD67hC0qAA813+iXNCcVSrDjBLMSdzZeixQtzf75zRN2VQ0TgVs0ZnuyZTJ5eLc5Lhuszh9Wr/6Hw==</diagram><diagram id="tbWY6QNJhf7gSaevbaGc" name="parse_ctx">7VjbctowEP0aT5/I+AKEPAZI0mlz66TTPGaEJWylsuRIMoR+fSVZvginDGQgl5lm8mAfr1bW7jm7a7xokj1fcJCnVwwi4oU+fPaiqReGQT8MPf3vw1WJjCog4Rhaowa4w3+QBX2LFhgi4RhKxojEuQvGjFIUSwcDnLOlazZnxN01BwnqAHcxIF30HkOZ2lMM/Ab/inCSVjsHvn2SgcrYAiIFkC1bUHTmRRPOmCyvsucJIjp4VVymU8Kuv9/iX9ntE2HnBeWzXq90dr7LkvoIHFH5atfX9080uEmfbi7pchaOryc/voleZI8mV1W8EFThs7eMy5QljAJy1qBjzgoKkfbqq7vG5pKxXIGBAh+RlCvLBVBIpqBUZsQ+RRSe6syqW8ooKpFzTIh1KSTgsrKAGGSMwgq2ZtrNllGx0ROs4DHaEAqbZbVJguQGO0t/HacWw2zMLxDLkOQrZcARARIvXBoCy+aktmsypi5s0nZIoH3rBSCF3QlIya9A7oVDoo4xnnF1lUgTrxLBFbBAsboEmcra2DyhIEPwIZe8hp2F6gVx42yNNy4rlimW6C4HJuJLVVlcBuyWugXiEj1vDLZ9GkU2HrZYBSf2ftlIP6j0nLZkP/QPlJ+qDL6twhwFWY215eO7KowJEALHjhC1GwhEal7jAHILt5Rbf99yM0vVycGqZZAzTKVoeb7VQEOswcglVhT6a9QoPTZEqV/t9dwJO9qulChyQB1SDZ8K3YbGSieyBwhOqBedKguC5voUJn2+EWVPlKrUTykz0qxXV+J+zB7apaASf7nrR9b/4MPpv/9eOdRtYFzM54i/phMIyTFNtmkD//L9Kfiy3i/Cd+fL8SdoFy8ObXtuD8Mt24P/IdpDcLzGo7doDyed0hJ0yKNriZtrjgT+A2bGQOfNHk1ZD8beYKorR1l6prHKpKoeujpwidUX1al9kGEIDfEImCEyBvHvxFBwwgjjZt9obv5epMUm2neUW38X2jf22p9eLynaPxpFw76TjKoG70aBTo6rdmy9Dt31bD4Xiqjr8t9DmkdbpPl/jdjHCNlQKBi5ya7H+MOWkXDzlLmrvVt2mtWH5+zwvaYexS2B+IMqGtrfg/xU00gQrHWR0XtPI0HQSSSWX4QRAiDEcNuEAuukqpIR+oVA+jLFwkSYF7EsuIYwFRIBqK7Y3K7TTjz9S1t+dHTUHSWX6ap2OLOjrI8WSDHkfD+dTlU5URa8oNX4DPW6bU/qSqkJHKvR+Kcpm6q0HHA47a+p+4XhtB5g23SIdqeDum1+yCyrQPNzcHT2Fw==</diagram><diagram id="6je8l0Wj5OvteJp03YQ2" name="callbacks">rZPBboMwDIafJsdKkEC160q7Vpq2SzdttyktbsgUCApuS/v0C2AKqJdNmpSD88Wxld9/mEjyeu1kmb3YFAzjQVozsWSchxHnrFlBeunIQw+U0yklDWCrr0AwIHrUKVSTRLTWoC6ncG+LAvY4YdI5e56mHayZdi2lgjuw3UtzTz90ihm9Ig4GvgGtsr5zGNBJLvtkAlUmU3seIbFiInHWYhfldQKmEa/XBXn4peX3vBLvm+T58zWqV9GsK/b0lyu3Jzgo8H9L0yxP0hxJrxxy6y6Mz41vtdg5H6kmMlYpcPfczw2hRpIIL73uLRSLDHPjQehDB5W+yl2bEPh9aXWB7XTjBYuXnsgj2qpzUHNBGq0KHxs4NKVO4FD7wT4SRlt6WpVyrwv11myWs8iTX6pFqjZVoR55hdRbg80BGyUCOuUxqUU/QfROOQ++mhPKRpaKiElysrpVHqblAxpYvx181Z6NfqdY/QA=</diagram></mxfile>
```

Comment:
This is a file containing some UML. I updated some of it, and it seems a new file format is being used why everything is changed.